### ISSUE TITLE
relative template paths and include_str support

### ISSUE THREAD

**Issue by henke443** [djc/askama#877]
I would want to have the feature.html template be in the same folder as feature.rs. 

It would be cool if you could do either one of:

A)

```rs

pub static FEATURE_TEMPLATE_HTML: &'static str = include_str!("feature.html");

#[derive(Template)]
#[template(source=FEATURE_TEMPLATE_HTML, ext = "html")] 
struct featureTemplate<'a> { 
    name: &'a str,
}
```

B)

```rs
#[derive(Template)] 
#[template(path="./feature.html")]  // And maybe just path="feature.html" is using the toml dirs? Like how "import" works in nodejs
struct featureTemplate<'a> { 
    name: &'a str, 
}
```

**Comment by djc** [comment]
I think I'm open to either of these, if you want take a whack at implementing this. Have a look at https://github.com/djc/askama/blob/main/askama_derive/src/generator.rs#L81 and https://github.com/djc/askama/blob/main/askama_derive/src/input.rs#L11.

**Comment by Kijewski** [comment]
The first option is impossible to implement. The proc macro will only see the token `FEATURE_TEMPLATE_HTML`, and it is currently not possible to evaluate the context. E.g. `#[doc = include_str()]` works, but `#[doc = SOME_CONSTANT]` does not.

But I like the second option!

**Comment by Zonnex** [comment]
Yes please! I REALLY wish to co-locate the html file with the rust file. I've actively looked for alternatives to askama that does this, but I haven't found any, and I don't want to use anything else because I really like how askama does things.
I doubt I'm able to implement this myself, I'll take a look but I otherwise that someone else is able to!

**Comment by BuriKizilkaya** [comment]
Is there any progress?

**Comment by djc** [comment]
No, I guess there isn't. Do you want to submit a PR?

**Comment by BuriKizilkaya** [comment]
I will try....

**Comment by andrewbaxter** [comment]
> The first option is impossible to implement. The proc macro will only see the token `FEATURE_TEMPLATE_HTML`, and it is currently not possible to evaluate the context. (E.g. that's why `#[doc = include_str()]` works, but `#[doc = SOME_CONSTANT]` does not.)
> 
> But I like the second option!

I think ATM the second option is also impossible to implement: https://github.com/rust-lang/rust/issues/54725 (going on 6 years)

`include_str` and I assume `doc` are currently using compiler magic to do it IIUC.

**Comment by joshka** [comment]
> i.e. I don't think there's a way to do this at the moment.

Use a nightly compiler, add cfg flags to your RUST_FLAGs and this PR does the trick:
https://github.com/djc/askama/pull/1112

Needs a bikeshed discussion on the feature flag name (as `relative-paths` is a little ambiguous for my liking), but it works.
